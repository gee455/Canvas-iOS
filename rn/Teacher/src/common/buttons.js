//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
 * @flow
 */

import React, { PureComponent } from 'react'
import {
  View,
  StyleSheet,
  TouchableOpacity,
} from 'react-native'
import BaseButton from 'react-native-button'
import colors from './colors'
import { Text, BOLD_FONT } from './text'

export function Button ({ style, containerStyle, ...props }: Object) {
  let brandingContainerStyles = {
    backgroundColor: colors.primaryButtonColor,
  }
  let brandingStyles = {
    color: colors.primaryButtonTextColor,
  }
  return (<BaseButton style={[styles.textColor, brandingStyles, style]} containerStyle={[styles.container, brandingContainerStyles, containerStyle]} {...props} />)
}

export class LinkButton extends PureComponent<Object> {
  render () {
    const {
      textStyle,
      ...props
    } = this.props
    const brandingStyles = { color: colors.primaryButtonColor }
    const textAttributes = this.props.textAttributes || {}
    const textStyles = [linkButtonStyles.textColor, linkButtonStyles.font, brandingStyles, textAttributes, textStyle].filter(Boolean)
    return (
      <TouchableOpacity
        hitSlop={{ top: 15, left: 15, bottom: 15, right: 15 }}
        {...props}
        accessibilityTraits='button'
      >
        <View>
          <Text style={textStyles}>
            {this.props.children}
          </Text>
        </View>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primaryButtonColor,
    overflow: 'hidden',
    padding: 10,
    borderRadius: 8,
  },
  textColor: {
    color: colors.primaryButtonTextColor,
  },
})

const linkButtonStyles = StyleSheet.create({
  font: {
    fontFamily: BOLD_FONT,
  },
  textColor: {
    fontSize: 14,
    color: colors.primaryButtonColor,
  },
})
