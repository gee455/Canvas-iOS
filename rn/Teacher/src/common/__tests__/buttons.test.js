//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

/**
 * @flow
 */

import React from 'react'
import { Button, LinkButton } from '../buttons'
import renderer from 'react-test-renderer'

jest.mock('react-native-button', () => 'Button')

test('renders button correctly', () => {
  let tree = renderer.create(
    <Button />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('renders link correctly', () => {
  let tree = renderer.create(
    <LinkButton />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
