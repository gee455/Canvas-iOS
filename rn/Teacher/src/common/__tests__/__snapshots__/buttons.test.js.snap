// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<Button
  containerStyle={
    Array [
      Object {
        "backgroundColor": "#374A59",
        "borderRadius": 8,
        "overflow": "hidden",
        "padding": 10,
      },
      Object {
        "backgroundColor": "#374A59",
      },
      undefined,
    ]
  }
  style={
    Array [
      Object {
        "color": "white",
      },
      Object {
        "color": "white",
      },
      undefined,
    ]
  }
/>
`;

exports[`renders link correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={
    Object {
      "bottom": 15,
      "left": 15,
      "right": 15,
      "top": 15,
    }
  }
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": ".SFUIDisplay",
          },
          Object {
            "color": "#2D3B45",
            "fontSize": 16,
          },
          Array [
            Object {
              "color": "#374A59",
              "fontSize": 14,
            },
            Object {
              "fontFamily": ".SFUIDisplay-bold",
            },
            Object {
              "color": "#374A59",
            },
            Object {},
          ],
          Object {
            "fontFamily": ".SFUIDisplay",
          },
        ]
      }
    />
  </View>
</View>
`;
